proxy/src/auth/credentials.rs

```diff
@@ -43,7 +43,7 @@ pub struct ClientCredentials {
     // TODO: this is a severe misnomer! We should think of a new name ASAP.
```

Comment:
This structure `ClientCredentials` feels kind of the same to `ComputeUserInfo`. 
If I am wrong, than it makes sense to rename some fields. Otherwise, that would be good to unify it.

Reply:
They are similar. The difference is that `endpoint` is optional in one, but required in the other.